File: src/main/java/org/dxworks/githubminer/dto/request/repository/pullrequests/CreatePullRequestBody.java
package org.dxworks.githubminer.dto.request.repository.pullrequests;

import com.google.api.client.util.Key;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class CreatePullRequestBody {
    @Key
    private String title;
    @Key
    private String head;
    @Key
    private String base;
    @Key
    private String body;
    @Key("maintainer_can_modify")
    private boolean maintainerCanModify;
    @Key
    private boolean draft;
}

File: src/main/kotlin/org/dxworks/githubminer/dto/request/repository/pullrequests/CreatePullRequestBody.kt
package org.dxworks.githubminer.dto.request.repository.pullrequests

import com.google.api.client.util.Key

class CreatePullRequestBody {
    @Key
    var title: String? = null

    @Key
    var head: String? = null

    @Key
    var base: String? = null

    @Key
    var body: String? = null

    @Key("maintainer_can_modify")
    var maintainerCanModify = false

    @Key
    var draft = false
}
